tests/test_main.cpp

```diff
@@ -54,40 +53,63 @@
 #include "tests/test_macros.h"
 
 int test_main(int argc, char *argv[]) {
+	bool run_tests = true;
+
+	// Convert arguments to Godot's command-line.
+	List<String> args;
+
+	for (int i = 0; i < argc; i++) {
+		args.push_back(String::utf8(argv[i]));
+	}
+	OS::get_singleton()->set_cmdline("", args);
```

Comment:
The first argument is empty because there's no way to know the executable name with the existing testing interface, but I'm not sure whether that's essential in the first place. The GDScript test tools cannot seem to recognize relative paths currently, maybe that's needed for this?